### PR TITLE
fix(langchain): Add trace context to client requests

### PR DESCRIPTION
**Pull request by ronensc**
This PR addresses part of issue #1983 by wrapping the `_generate()`, `_agenerate()`, `_stream()`, and `_astream()` methods in the OpenAI classes to propagate the trace context through the request's HTTP headers.

However, the current implementation does not fully support streaming for `ChatModels`. The logic there differs slightly, and I was unable to access the `run_id` within `BaseChatOpenAI._stream()`. The `run_id` is essential for matching the correct span, and in other methods, I retrieve it via the `run_manager`. While `BaseChatOpenAI._stream()` accepts a `run_manager` parameter, it is not being passed in this case.

For reference:
- [BaseChatOpenAI._stream() signature](https://github.com/langchain-ai/langchain/blob/fc14f675f1af1e8fee23710f208259da9fcfd778/libs/partners/openai/langchain_openai/chat_models/base.py#L612)
- [Related section in chat_models.py](https://github.com/langchain-ai/langchain/blob/fc14f675f1af1e8fee23710f208259da9fcfd778/libs/core/langchain_core/language_models/chat_models.py#L385-L400)

The code that wraps the streaming methods `_stream()` and `_astream()` of the `ChatModels` is currently commented out, and the relevant tests are marked with `xfail()` to indicate their expected failure.

- [x] I have added tests that cover my changes.
- [ ] If adding a new instrumentation or changing an existing one, I've added screenshots from some observability platform showing the change.
- [ ] PR name follows conventional commits format: `feat(instrumentation): ...` or `fix(instrumentation): ...`.
- [ ] (If applicable) I have updated the documentation accordingly.

### CODE REVIEW

File: packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py [hunk 1]
```diff
@@ -63,3 +163,40 @@ def __call__(
...
+        if run_manager:
+            run_id = run_manager.run_id
+            span_holder = self._callback_manager.spans[run_id]
+
+            extra_headers = kwargs.get("extra_headers", {})
+
+            # Inject tracing context into the extra headers
+            ctx = set_span_in_context(span_holder.span)
+            TraceContextTextMapPropagator().inject(extra_headers, context=ctx)
+            # TraceContextTextMapPropagator().inject(extra_headers, context=span_holder.context)
```
Comment (nirga) [inline packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py:197]: nit: remove?

File: packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py [hunk 2]
```diff
@@ -36,14 +41,109 @@ def _instrument(self, **kwargs):
...
+
+        if not self.disable_trace_context_propagation:
+            self._wrap_openai_functions_for_tracing(traceloopCallbackHandler)
+
+    def _wrap_openai_functions_for_tracing(self, traceloopCallbackHandler):
+        openai_tracing_wrapper = _OpenAITracingWrapper(traceloopCallbackHandler)
+
+        # Wrap langchain_community.llms.openai.BaseOpenAI
+        wrap_function_wrapper(
+            module="langchain_community.llms.openai",
```
Comment (nirga) [inline packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py:59]: Why not wrap the base class `BaseLLM`? https://github.com/langchain-ai/langchain/blob/316541536940a0e063ed09f48d2bec05ded862c3/libs/core/langchain_core/language_models/llms.py#L292
Then theoretically, it should work for all models

**Review by nirga — COMMENTED** [review]
Thanks so much @ronensc! Really liked the through testing!

File: packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py (hunk 2 above)
Comment (ronensc) [reply packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py:59]: I agree that wrapping the base class would be preferable. However, I couldn't find a way to access the `run_id`. For example, when I set a breakpoint in the `invoke()` method while debugging `test_trace_propagation[OpenAI]`, I found that `config["run_id"]` is [empty](https://github.com/langchain-ai/langchain/blob/316541536940a0e063ed09f48d2bec05ded862c3/libs/core/langchain_core/language_models/llms.py#L397). The `run_id` is determined further down the stack trace.

Comment (nirga) [reply packages/opentelemetry-instrumentation-langchain/opentelemetry/instrumentation/langchain/__init__.py:59]: Oh well
It's a good start though

**Review by nirga — APPROVED** [review]

**Pull request merged**